packages/trpc/server/trpc.ts

```diff
@@ -41,10 +43,41 @@ const isAdminMiddleware = t.middleware(({ ctx, next }) => {
     },
   });
 });
+interface IRateLimitOptions {
+  intervalInMs: number;
+  limit: number;
+}
+const isRateLimitedByUserIdMiddleware = ({ intervalInMs, limit }: IRateLimitOptions) =>
+  t.middleware(({ ctx, next }) => {
+    // validate user exists
+    if (!ctx.user) {
+      throw new TRPCError({ code: "UNAUTHORIZED" });
+    }
```

Comment:
Tagging @KATT to give some insight. Is there a way to avoid having to re-check this on every middleware? As you can see we already have an `isAuthedMiddleware` but when using the procedure the `ctx.user` inference is lost.

Reply:
https://github.com/calcom/cal.com/pull/6017/files#r1047771652